runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -0,0 +1,203 @@
+        let acct = self.basic(address);
+
+        !(acct.nonce == primitive_types::U256::zero()
+            && acct.balance == primitive_types::U256::zero())
+    }
+
+    fn basic(&self, address: primitive_types::H160) -> Basic {
+        let mut ctx = self.ctx.borrow_mut();
+        let mut state = ctx.runtime_state();
+
+        // Derive SDK account address from the Ethereum address.
+        let address = Cfg::map_address(address);
+        // Fetch balance and nonce from SDK accounts.
+        let balance = Cfg::Accounts::get_balance(&mut state, address, Cfg::TOKEN_DENOMINATION)
+            .unwrap_or_default();
+        let nonce = Cfg::Accounts::get_nonce(&mut state, address).unwrap_or_default();
+
+        Basic {
+            nonce: nonce.into(),
+            balance: balance.into(),
+        }
+    }
+
+    fn code(&self, address: primitive_types::H160) -> Vec<u8> {
+        let address: H160 = address.into();
+
+        let mut ctx = self.ctx.borrow_mut();
+        let store = state::codes(ctx.runtime_state());
+        store.get(&address).unwrap_or_default()
+    }
+
+    fn storage(
+        &self,
+        address: primitive_types::H160,
+        index: primitive_types::H256,
+    ) -> primitive_types::H256 {
+        let address: H160 = address.into();
+        let idx: H256 = index.into();
+
+        let mut ctx = self.ctx.borrow_mut();
+        let store = state::storage(ctx.runtime_state(), &address);
+        let res: H256 = store.get(&idx).unwrap_or_default();
+        res.into()
+    }
+
+    fn original_storage(
+        &self,
+        _address: primitive_types::H160,
+        _index: primitive_types::H256,
+    ) -> Option<primitive_types::H256> {
+        None
+    }
+}
+
+impl<'c, C: Context, Cfg: Config> ApplyBackend for Backend<'c, C, Cfg> {
+    fn apply<A, I, L>(&mut self, values: A, logs: L, _delete_empty: bool)
+    where
+        A: IntoIterator<Item = Apply<I>>,
+        I: IntoIterator<Item = (primitive_types::H256, primitive_types::H256)>,
+        L: IntoIterator<Item = Log>,
+    {
+        for apply in values {
+            match apply {
+                Apply::Modify {
+                    address,
+                    basic,
+                    code,
+                    storage,
+                    reset_storage,
+                } => {
+                    let addr: H160 = address.into();
+                    // Derive SDK account address from the Ethereum address.
+                    let address = Cfg::map_address(address);
+
+                    // Update account balance and nonce.
+                    let mut state = self.ctx.get_mut().runtime_state();
+                    let amount =
+                        token::BaseUnits::new(basic.balance.as_u128(), Cfg::TOKEN_DENOMINATION);
+                    Cfg::Accounts::set_balance(&mut state, address, &amount);
+                    Cfg::Accounts::set_nonce(state, address, basic.nonce.as_u64());
```

Comment:
These `{get,set}_{balance,nonce}` calls seem to always appear in pairs -- would it make more sense to join them into a `{get,set}_balance_and_nonce` call?  This would save one load and one store operation from/to storage.